Hudl.Mjolnir/Command/Command.cs

```diff
@@ -95,17 +96,24 @@ internal IStats Stats
             set { _stats = value; }
         }
 
+        private IMetricEvents _metricEvents;
+        internal IMetricEvents MetricEvents
+        {
+            private get { return _metricEvents ?? CommandContext.MetricEvents; }
+            set { _metricEvents = value; }
+        }
+
         private ICircuitBreaker _breaker;
         internal ICircuitBreaker CircuitBreaker
         {
-            private get { return _breaker ?? CommandContext.GetCircuitBreaker(_breakerKey); }
+            private get { return _breaker ?? CommandContext.Current.GetCircuitBreaker(_breakerKey); }
```

Comment:
True - it's kind of a baked-in assumption within the library. I'd like to rework the idea of `CommandContext` more in the future and make things more injectable.